### PR TITLE
QE: Fix Prometheus issues in the testsuite

### PR DESCRIPTION
**Pull request by nodeg**
## What does this PR change?

This will enable (and afterwards disable) the tools update repository on the SLE minion in order to be able to install the Prometheus package (`golang-github-prometheus-prometheus`).

Fixes our Prometheus issue in the CI we see on Uyuni, head and 4.2.

image

## GUI diff

No difference.

- [x] **DONE**

## Documentation

- No documentation needed: only internal and user invisible changes

- [x] **DONE**

- Cucumber tests were adjusted

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/17512

Manager 4.2
Manager 4.1

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by nodeg** [comment]
We also have those steps for a CentOS Salt minion. Should I adjust this feature, too?

**Comment by ktsamis** [comment]
We should merge them in one scenario yes. Not sure if this is a WORKAROUND or a prerequisite that we were lacking. Unless we add the packages in bootstrap repo or make a change in the product we will be keeping this for a while I think

**Review by srbarrios — APPROVED** [review]

**Comment by nodeg** [comment]
https://ci.suse.de/job/uyuni-prs-ci-tests/1286

```bash
23:10:44  @sle_minion @scope_monitoring
23:10:44  Feature: Monitor SUMA environment with Prometheus on a SLE Salt minion
23:10:44  In order to monitor Uyuni server
23:10:44  As an authorized user
23:10:44  I want to enable Prometheus exporters
23:10:44  
23:10:44  Scenario: Pre-requisite: enable Prometheus exporters repository on the minion                          # features/secondary/min_monitoring.feature:14
23:10:44  This scenario ran at: 2022-04-21 23:10:31 +0200
23:10:44  Node: suma-pr7-min-sles15, OS Version: 15-SP3, Family: sles
23:10:44  When I enable the necessary repositories before installing Prometheus exporters on this "sle_minion" # features/step_definitions/command_steps.rb:1523
23:10:44        FAIL: zypper mr --enable  ||:; zypper mr --enable  os_pool_repo os_update_repo tools_pool_repo tools_update_repo; returned status code = 3.
23:10:44        Output:
23:10:44        Alias or an aggregate option is required.
23:10:44        modifyrepo (mr) <OPTIONS> <ALIAS|#|URI>
23:10:44        modifyrepo (mr) <OPTIONS> <--all|--remote|--local|--medium-type>
23:10:44        
23:10:44        Modify properties of repositories specified by alias, number, or URI, or by the '--all, --remote,
23:10:44        --local, --medium-type' aggregate options.
23:10:44        
23:10:44          Command options:
23:10:44        
23:10:44        -n, --name <NAME>           Set a descriptive name for the repository.
23:10:44        -e, --enable                Enable a disabled repository. Default: true
23:10:44        -d, --disable               Disable the repository (but don't remove it).
23:10:44        -f, --refresh               Enable auto-refresh of the repository. Default: false
23:10:44        -F, --no-refresh            Disable auto-refresh of the repository.
23:10:44        -p, --priority <PRIORITY>   Set priority of the repository. Default: 0
23:10:44        -k, --keep-packages         Enable RPM files caching. Default: false
23:10:44        -K, --no-keep-packages      Disable RPM files caching.
23:10:44        -g, --gpgcheck              Enable GPG check for this repository.
23:10:44            --gpgcheck-strict       Enable strict GPG check for this repository.
23:10:44            --gpgcheck-allow-unsigned
23:10:44                                    Short hand for '--gpgcheck-allow-unsigned-repo
23:10:44                                    --gpgcheck-allow-unsigned-package'.
23:10:44            --gpgcheck-allow-unsigned-repo
23:10:44                                    Enable GPG check but allow the repository metadata to be unsigned.
23:10:44            --gpgcheck-allow-unsigned-package
23:10:44                                    Enable GPG check but allow installing unsigned packages from this
23:10:44                                    repository.
23:10:44        -G, --no-gpgcheck           Disable GPG check for this repository.
23:10:44            --default-gpgcheck      Use the global GPG check setting defined in /etc/zypp/zypp.conf. This is
23:10:44                                    the default.
23:10:44        -a, --all                   Apply changes to all repositories. Default: false
23:10:44        -l, --local                 Apply changes to all local repositories. Default: false
23:10:44        -t, --remote                Apply changes to all remote repositories. Default: false
23:10:44        -m, --medium-type <TYPE>    Apply changes to repositories of specified type.
23:10:44        
23:10:44          Legacy options:
23:10:44        
23:10:44        -r                          The same as -f.
23:10:44        -R                          The same as -F.
23:10:44        
23:10:44        Repository 'os_pool_repo' has been successfully enabled.
23:10:44        Repository 'os_update_repo' has been successfully enabled.
23:10:44        Repository 'tools_pool_repo' has been successfully enabled.
23:10:44        Repository tools_update_repo not found.
23:10:44         (RuntimeError)
23:10:44        ./features/support/lavanda.rb:75:in `run'
23:10:44        ./features/step_definitions/command_steps.rb:850:in `/^I (enable|disable) (the repositories|repository) "([^"]*)" on this "([^"]*)"((?: without error control)?)$/'
23:10:44        ./features/step_definitions/command_steps.rb:1525:in `/^I (enable|disable) the necessary repositories before installing Prometheus exporters on this "([^"]*)"((?: without error control)?)$/'
23:10:44        features/secondary/min_monitoring.feature:15:in `I enable the necessary repositories before installing Prometheus exporters on this "sle_minion"'
23:10:44  And I run "zypper -n ref" on "sle_minion" 
```
It seems that the repository `tools_update_repo` does not exist on this minion.

Furthermore we have another error for the CentOS minion:

```bash
23:08:22  @centos_minion @scope_monitoring @scope_res
23:08:22  Feature: Monitor SUMA environment with Prometheus on a CentOS Salt minion
23:08:22  In order to monitor Uyuni server
23:08:22  As an authorized user
23:08:22  I want to enable Prometheus exporters
23:08:22  
23:08:22  Scenario: Pre-requisite: enable Prometheus exporters repository on the CentOS minion                    # features/secondary/min_centos_monitoring.feature:15
23:08:22  This scenario ran at: 2022-04-21 23:08:09 +0200
23:08:22  Node: suma-pr7-min-centos7, OS Version: 7, Family: centos
23:08:22  When I enable the necessary repositories before installing Prometheus exporters on this "ceos_minion" # features/step_definitions/command_steps.rb:1523
23:08:22        FAIL: sed -i 's/enabled=.*/enabled=1/g' /etc/yum.repos.d/os_pool_repo.repo; sed -i 's/enabled=.*/enabled=1/g' /etc/yum.repos.d/os_update_repo.repo; sed -i 's/enabled=.*/enabled=1/g' /etc/yum.repos.d/tools_pool_repo.repo; sed -i 's/enabled=.*/enabled=1/g' /etc/yum.repos.d/tools_update_repo.repo;  returned status code = 2.
23:08:22        Output:
23:08:22        sed: can't read /etc/yum.repos.d/os_pool_repo.repo: No such file or directory
23:08:22        sed: can't read /etc/yum.repos.d/os_update_repo.repo: No such file or directory
23:08:22        sed: can't read /etc/yum.repos.d/tools_update_repo.repo: No such file or directory
23:08:22         (RuntimeError)
23:08:22        ./features/support/lavanda.rb:75:in `run'
23:08:22        ./features/step_definitions/command_steps.rb:850:in `/^I (enable|disable) (the repositories|repository) "([^"]*)" on this "([^"]*)"((?: without error control)?)$/'
23:08:22        ./features/step_definitions/command_steps.rb:1525:in `/^I (enable|disable) the necessary repositories before installing Prometheus exporters on this "([^"]*)"((?: without error control)?)$/'
23:08:22        features/secondary/min_centos_monitoring.feature:16:in `I enable the necessary repositories before installing Prometheus exporters on this "ceos_minion"'
```

**Comment by nodeg** [comment]
@rjmateus according to @srbarrios it seems we do not have that repository available for this `product_version` and module name in sumaform.

**Comment by witekest** [comment]
In case of `suma-head-min-sles15` I was enabling `tools_additional_repo` repository.
https://github.com/SUSE/spacewalk/issues/17512#issuecomment-1098097005

**Comment by nodeg** [comment]
Thanks Witek, I will change that and do another CI run.

File: testsuite/features/secondary/min_monitoring.feature [hunk 1]
```diff
@@ -48,6 +48,16 @@ Feature: Monitor SUMA environment with Prometheus on a SLE Salt minion
     And I click on "Save"
     Then I should see a "Formula saved" text
 
+  @susemanager
+  Scenario: Apply highstate for Prometheus exporters
+    When I enable the repositories "tools_additional_repo tools_additional_repo" on this "sle_minion" without error control
```
Comment (srbarrios) [inline testsuite/features/secondary/min_monitoring.feature:53]: Type? Two times same repo

File: testsuite/features/secondary/min_monitoring.feature [hunk 2]
```diff
@@ -48,6 +48,16 @@ Feature: Monitor SUMA environment with Prometheus on a SLE Salt minion
...
+    When I follow "States" in the content area
+    And I click on "Apply Highstate"
+    Then I should see a "Applying the highstate has been scheduled." text
+    And I wait until event "Apply highstate scheduled by admin" is completed
+    Then I disable the repositories "tools_additional_repo tools_additional_repo" on this "sle_minion" without error control
```
Comment (srbarrios) [inline testsuite/features/secondary/min_monitoring.feature:58]: Same

File: testsuite/features/secondary/min_monitoring.feature (hunk 1 above)
Comment (nodeg) [reply testsuite/features/secondary/min_monitoring.feature:53]: I just followed your recommendation from here: https://github.com/SUSE/spacewalk/issues/17512#issuecomment-1105205452

Comment (nodeg) [reply testsuite/features/secondary/min_monitoring.feature]: Fixed.

File: testsuite/features/secondary/min_monitoring.feature (hunk 2 above)
Comment (nodeg) [reply testsuite/features/secondary/min_monitoring.feature]: Fixed.

**Review by srbarrios — APPROVED** [review]
I'm not sure if tools_pool_repo should still be enabled in both cases.. but let's merge this, if it fixes it, then all good for now. 

Basically I don't know if at some point we will not have any content in the updates but the content will be in the pool, it happens when we have a new version of a product, for instance in sp4 we will probably not have content in updates but in pool. But honestly, when it comes to client tools, I'm not that sure about it.

**Comment by nodeg** [comment]
Ok, the error stays the same as before:

```bash
16:11:55  Scenario: Pre-requisite: enable Prometheus exporters repository on the minion                          # features/secondary/min_monitoring.feature:14
16:11:55  This scenario ran at: 2022-04-22 16:11:40 +0200
16:11:55  Node: suma-pr2-min-sles15, OS Version: 15-SP3, Family: sles
16:11:55  When I enable the necessary repositories before installing Prometheus exporters on this "sle_minion" # features/step_definitions/command_steps.rb:1523
16:11:55        FAIL: zypper mr --enable  ||:; zypper mr --enable  os_pool_repo os_update_repo tools_pool_repo tools_update_repo; returned status code = 3.
16:11:55        Output:
(...)
16:11:55        Repository 'os_pool_repo' has been successfully enabled.
16:11:55        Repository 'os_update_repo' has been successfully enabled.
16:11:55        Repository 'tools_pool_repo' has been successfully enabled.
16:11:55        Repository tools_update_repo not found.
16:11:55         (RuntimeError)
16:11:55        ./features/support/lavanda.rb:75:in `run'
16:11:55        ./features/step_definitions/command_steps.rb:850:in `/^I (enable|disable) (the repositories|repository) "([^"]*)" on this "([^"]*)"((?: without error control)?)$/'
16:11:55        ./features/step_definitions/command_steps.rb:1525:in `/^I (enable|disable) the necessary repositories before installing Prometheus exporters on this "([^"]*)"((?: without error control)?)$/'
16:11:55        features/secondary/min_monitoring.feature:15:in `I enable the necessary repositories before installing Prometheus exporters on this "sle_minion"'
16:11:55  And I run "zypper -n ref" on "sle_minion" 
```

Does it make sense to merge it then?

**Comment by nodeg** [comment]
A full run of the testsuite did not solve the problem. See run [#1293](https://ci.suse.de/view/Manager/view/Uyuni/job/uyuni-prs-ci-tests/1293/).

**Comment by nodeg** [comment]
Not needed anymore.

**Pull request closed without merging**